api/api.go

```diff
@@ -171,8 +173,8 @@ func (api *ExtensionsApi) getMergedExtensionsMap(extensions []map[string]interfa
 	results := make([]map[string]interface{}, 0)
 	for _, extension := range api.Extensions {
 		if target, found := targetExtensions[extension.UUID]; found {
-			extensionWithUrls := setExtensionUrls(extension, rootUrl)
-			extensionData, err := interfaceToMap(extensionWithUrls)
+			extensionWithRuntimeData := setExtensionWithRuntimeData(extension, rootUrl)
```

Comment:
Can we make this more performant? Currently this will run every time an update has been made to the extension even nothing has changed in the locales. I think we need to update extension data directly. The current behaviour for `setExtensionUrls` is only needed with every notification and API response because the url changes depending on whether it's accessed via a tunnel or not.

I think we need to make the following changes:

1. In `main.go` add a go routine to start a file watcher for the `locales` directory when running `build.watch`: https://github.com/Shopify/shopify-cli-extensions/blob/4d5a8625fd84da9346f0075cee3d43a9a33ce788/main.go#L97
This will start the watcher every time `shopify extension serve` is run.

3. When the watcher detects change to the locales, merge the JSON data and then call 
```
api.Notify([]core.Extension{extension})
```
`api.Notify` takes care of merging the data and notifying all subscribers for you.

Reply:
@vividviolet sounds good! We'll move the `GetLocalization` function as you suggested for now. For the purpose of this PR, `GetLocalization` will only happen in response to requests to `/extentions/`. 

Thank you for your note on `fsnotify`, I had come across this in research and wasn't sure if this was something we could use. I will open up another PR for a patch to implement the file watcher. 

@kumar303 one thing to note, is we will still need to do work in `checkout-web` to actually refresh the translations from the websocket update.